chore(refactor): guide updates for metadata preparation (#1235)
lukso-network/docs: docs/learn/digital-assets/metadata-management/metadata-preparation.md

> To add metadata to your contract, you have to follow these steps:

Reviewer: ```suggestion
To set the metadata of a digital asset, you have to follow these steps:
```

1. Upload each media file (icon, picture, image, video, etc) and get it's `URLs` or `IPFS CID`.

> 2. Use the file hashes and URLs to generate the final [LSP4 Metadata JSON file](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-4-DigitalAsset-Metadata.md)

Reviewer: ```suggestion
2. Inside the [LSP4 Metadata JSON file](https://github.com/lukso-network/LIPs/blob/main/LSPs/LSP-4-DigitalAsset-Metadata.md#lsp4metadata), set the following values under each `"images"`:
    2.a. under `verification.data`, set the hash of the media file.
    2.b. under `url`, set the value `ipfs://<IPFS CID>` (obtained from step 1)
```

| Centralized Storage   | Centralized storage solutions are user-friendly and can be a good starting point for those new to the ecosystem or participating in hackathons. However, they rely on a single point of control, which may not align with the decentralized ethos of blockchain technology. | [AWS S3](https://aws.amazon.com/s3/), [Google Cloud](https://cloud.google.com/storage?hl=en), [Dropbox](https://www.dropbox.com/), etc. |
| Decentralized Storage | Decentralized storage solutions align with the decentralized nature of blockchain and offer benefits like _redundancy_, _censorship resistance_, and _permanent storage_. However, they might be more complex to interact with.                                             | [IPFS](https://ipfs.tech/), [Arweave](https://www.arweave.org/), [Filecoin](https://filecoin.io/), etc.                                 |

:::info Uploading files to IPFS

> You can visit our [Use IPFS Storage](../../other-guides/utilize-ipfs-storage.md) guide to upload your files to IPFS using the LUKSO [`tools-data-providers`](https://github.com/lukso-network/tools-data-providers) library. To upload your metadata to IPFS using the Cascade and Sense Protocol, you can visit [Cascade and Sense Support](./cascade-and-sense-support.md) guide.

Reviewer: Suggest shortening

```suggestion
See our guide [**"Use IPFS Storage"**](../../other-guides/utilize-ipfs-storage.md) to learn how to upload your files to IPFS using the LUKSO [`tools-data-providers`](https://github.com/lukso-network/tools-data-providers) library. To upload your metadata to IPFS using the Cascade and Sense Protocol, you can visit [Cascade and Sense Support](./cascade-and-sense-support.md) guide.
```

:::

## 2 - Generate the JSON File

The JSON content is used as input for encoding the metadata according to the [LSP4](/standards/tokens/LSP4-Digital-Asset-Metadata.md#lsp4---digital-asset-metadata) standard. After you filled your metadata file with content, upload it to a preferred storage solution.

To ensure the authenticity of the images, please **generate the hash** of the uploaded files and set them within the `"verification"` field of the JSON Metadata:

- Define **the used hash function** within the `"method"` element

> - Add and **generated hexadecimal hash** within the `"data"` element

Reviewer: Suggest precisions

```suggestion
- Add the **hash digest** (_e.g: 32 bytes long for keccak256_) within the `"data"` element
```
